pkg/rbd/controllerserver.go

```diff
@@ -93,7 +93,25 @@ func (cs *ControllerServer) validateVolumeReq(req *csi.CreateVolumeRequest) erro
 func parseVolCreateRequest(req *csi.CreateVolumeRequest) (*rbdVolume, error) {
 	// TODO (sbezverk) Last check for not exceeding total storage capacity
 
-	rbdVol, err := getRBDVolumeOptions(req.GetParameters())
+	isMultiNode := false
+	isBlock := false
+	for _, cap := range req.VolumeCapabilities {
+		// Only checking SINGLE_NODE_SINGLE_WRITER here because regardless of the other types (MULTI READER) we need to implement the same logic to ignore the in-use response
```

Comment:
This will work with today's advertised capabilities but it is fragile if future capabilites are added in k8s.  SINGLE_NODE_READER_ONLY is not MultiNode.  MULTI_NODE_READER_ONLY & MULTI_NODE_SINGLE_WRITER may be safe because there is no concurrent writing going on.

Would you consider testing explicitly for MULTI_NODE_MULTI_WRITER doing the fail early at line 108ff if that is true ?

Reply:
The thing is it's not a check of whether it's "safe" or not, it's a check of whether K8s and the plugin will let you even request the mode.  I'm not sure I follow the "fragile" bit?  We're explicitly supporting a single capability here, if other capabilities are added to K8s or if we choose to support additional capabilities in the plugin we should do that explicitly.

By the way, one of the reasons I thought about this and changed it was because you raised concerns about not being explicit and I was using an "any multi-node" logic; so I'm sort of confused.  Let me know if I'm missing something; I might just not be seeing something.

Reply:
I realize I didn't express my concern well and that what bothers me here is the name of the flag ``is_multinode``.  One of the access_modes that isn't SINGLE_MODE_SINGLE_WRITER is SINGLE_NODE_SINGLE_READER so it seems wrong to say ``is_multinode`` would be true.   In fact SINGLE_NODE_SINGLE_READER should be safe w/o block mode too, right?